### PR TITLE
allow override implementations on Nodes and Links #3

### PR DESCRIPTION
**Pull request by rarous**
Instead of type dispatch this uses property based dispatch. Node is every child with `node` prop. Link is every child with `link` prop.

### CODE REVIEW

File: src/utils/sets-equal.js [hunk 1]
```diff
@@ -30,11 +30,10 @@ export default function setsEqual(setA, setB) {
...
-      return false;
-    }
-  }
+  let acc = true;
+  setA.forEach((a) => {
+    acc = acc && setB.has(a);
+  });
 
-  return true;
+  return acc;
```
Comment (fastfrwrd) [inline src/utils/sets-equal.js]: why did this need to be updated?

**Comment by fastfrwrd** [comment]
We need to add a story to the docs to demonstrate this working, I'm happy to do that outside the scope of this PR.

File: src/utils/sets-equal.js (hunk 1 above)
Comment (rarous) [inline src/utils/sets-equal.js]: I see the PR includes moře commits from my Fork, it should have only the First oné.

**Comment by fastfrwrd** [comment]
Thanks for the PR! Fixes #3.

**Pull request merged**